tape/src/test/java/com/squareup/tape/QueueFileTest.java

```diff
@@ -958,6 +960,79 @@ public void testForEachReadWithOffset() throws IOException {
     assertThat(queueFile2.size()).isEqualTo(queueSize);
   }
 
+  @Test public void testIterator() throws IOException {
+    byte[] data = values[10];
+
+    for (int i = 0; i < 10; i++) {
+      QueueFile queueFile = new QueueFile(file);
+      for (int j = 0; j < i; j++) {
+        queueFile.add(data);
+      }
+
+      int saw = 0;
+      for (byte[] element : queueFile) {
+        assertThat(element).isEqualTo(data);
+        saw++;
+      }
+      assertThat(saw).isEqualTo(i);
+      file.delete();
+    }
+  }
+
+  @Test public void testIteratorNext() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+    queueFile.add(values[10]);
+  }
+
+  @Test public void testIteratorRemove() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+    for (int i = 0; i < 15; i++) {
+      queueFile.add(values[i]);
+    }
+
+    Iterator<byte[]> iterator = queueFile.iterator();
+    while (iterator.hasNext()) {
+      iterator.next();
+      iterator.remove();
+    }
+
+    assertThat(queueFile).isEmpty();
+  }
+
+  @Test public void testIteratorRemoveDisallowsConcurrentModification() throws IOException {
+    QueueFile queueFile = new QueueFile(file);
+    for (int i = 0; i < 15; i++) {
+      queueFile.add(values[i]);
+    }
+
+    Iterator<byte[]> iterator = queueFile.iterator();
+    iterator.next();
+    queueFile.remove();
+    try {
+      iterator.remove();
+      fail("remove should not permit concurrent modifications.");
```

Comment:
can you switch the import to `org.junit.Assert.fail` and remove these two messages? If this fails the reason is already clear and I don't like being forced to provide a redundant message.